pkg/resource/patch.go

```diff
@@ -27,11 +27,7 @@ func (r *helper) Patch(name types.NamespacedName, kind, apiVersion string, patch
 		Out:    &bytes.Buffer{},
 		ErrOut: &bytes.Buffer{},
 	}
-	factory, err := r.getFactory(name.Namespace)
```

Comment:
This is where I got into trouble, I didn't notice that some of these getFactory commands were specifying a namespace, so my attempt to store the factory/dynClient/etc on the helper and re-use them was misguided. I'm reworking it to only cache the openapi schema.